src/main/java/org/tikv/common/util/ChannelFactory.java

```diff
@@ -361,7 +364,7 @@ public void close() {
     connPool.clear();
 
     if (certContext != null) {
-      recycler.shutdown();
+      recycler.shutdownNow();
```

Comment:
What about removing the `if (certContext != null)`:

```java
if (recycler != null ) {
    // shutdown recycler
}

if (certWatcher != null) {
    // shutdown certWatcher
}
```

Reply:
@iosmanthus Thanks for your review. OK, I will change it.